src/NuGet.Core/NuGet.Configuration/PackageNamespaces/SearchNode.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Configuration
+{
+    internal class SearchNode
+    {
+        public readonly SearchNode Parent;
+        public readonly Dictionary<char, SearchNode> Children;
+        public bool IsValueNode => !string.IsNullOrWhiteSpace(NamespaceId);
+        public bool IsGlobbing { get; set; }
```

Comment:
I think we can the eliminate `IsGlobbing` property but it depends. In `fullySpecified` mode, which source are we supposed to pick for `System.Text` package because this package Id matches `System.Text*` namespace defined on `Source2`?
``` xml
<packageNamespaces>
    <packageSource key="Source1">
      <namespace id="System.Text" />
    </packageSource>
    <packageSource key="Source2">
      <namespace id="System.Text*" />
    </packageSource>
  </packageNamespaces>
```

Reply:
In this case it matches both of them but in `strict` mode it'll throw exception.

Reply:
I don't think we can remove is globbing, it's useful for distinguish `Hello.` vs `Hello.*` vs `Hello` vs `Hello*`.